sos/__init__.py

```diff
@@ -39,325 +36,99 @@ def _default(msg):
+                            help="specify alternate configuration file")
+        parser.add_argument("--debug", action="store_true", dest="debug",
+                            help="enable interactive debugging using the "
+                            "python debugger")
+        parser.add_argument("-q", "--quiet", action="store_true",
+                            dest="quiet", default=False,
+                            help="only print fatal errors")
+        parser.add_argument("-s", "--sysroot", action="store", dest="sysroot",
+                            help="system root directory path (default='/')",
+                            default=None)
+        parser.add_argument("--tmp-dir", action="store",
+                            dest="tmp_dir",
+                            help="specify alternate temporary directory",
+                            default=None)
+        parser.add_argument("-t", "--threads", action="store", dest="threads",
+                            default=4, type=int,
+                            help="Number of threads to use")
+        parser.add_argument("-v", "--verbose", action="count",
+                            dest="verbosity", default=0,
+                            help="increase verbosity")
+
+    def _init_component(self):
+        """Determine which component has been requested by the user, and then
+        initialize that component.
         """
-        def has_value(name, value):
-            """ Test for non-null option values.
-            """
-            null_values = ("False", "None", "[]", '""', "''", "0")
-            if not value or value in null_values:
-                return False
-            if name in _arg_defaults:
-                if str(value) == str(_arg_defaults[name]):
-                    return False
-            return True
-
-        def filter_opt(name, value):
-            """ Filter out preset and null-valued options.
-            """
-            if name in ("add_preset", "del_preset", "desc", "note"):
-                return False
-            return has_value(name, value)
-
-        def argify(name, value):
-            """ Convert sos option notation to command line arguments.
-            """
-            # Handle --verbosity specially
-            if name.startswith("verbosity"):
-                arg = "-" + int(value) * "v"
-                return arg
-
-            name = name.replace("_", "-")
-
-            value = ",".join(value) if _is_seq(value) else value
-
-            if value is not True:
-                opt = "%s %s" % (name, value)
-            else:
-                opt = name
-
-            arg = "--" + opt if len(opt) > 1 else "-" + opt
-            return arg
-
-        opt_items = sorted(self.dict().items(), key=lambda x: x[0])
-        return [argify(n, v) for (n, v) in opt_items if filter_opt(n, v)]
+        _com = self.args.component
+        if _com not in self._components.keys():
```

Comment:
That's the reasoning behind `set_defaults()` on the subparsers - so that `_com` will always be set to the "normal" name of the command instead of an alias, so we only need to check the dict keys within `self._components`.